pkgs/os-specific/bsd/netbsd/default.nix

```diff
@@ -115,30 +135,33 @@ let
 
       runHook postInstall
     '';
-    extraPaths = [ make.src ] ++ make.extraPaths;
+    extraPaths = with self; [ make.src ] ++ make.extraPaths;
   };
 
-  compat = if stdenv.hostPlatform.isNetBSD then stdenv else mkDerivation rec {
+  compat = if stdenvNoCC.hostPlatform.isNetBSD then stdenvNoCC else mkDerivation (let
+    version = "8.0";
+    commonDeps = [ zlib ];
```

Comment:
I don't think I understand why we need zlib in both sets of dependencies.  The manual says "If the dependency doesn’t care about the target platform (i.e. isn’t a compiler or similar tool), put it here [in `nativeBuildInputs`]".

Reply:
Well, the manual is simplifying here :). libraries being fed tools in `depsBuildBuild` go in `nativeBuildInputs` (or `depsBuildBuild` or `depsBuildTarget`) so the second `Build` of the tool lines up with the first `Build` (`nativeBuildInputs = `depsBuildHost`) of the library.

In this case I was just going by the comment; I have no idea whether we need to run a freshly-built throwaway tool that links zlib and I suspect the answer is "no".